RecoCTPPS/TotemRPLocal/src/CTPPSDiamondTrackRecognition.cc

```diff
@@ -0,0 +1,146 @@
+/****************************************************************************
+ *
+ * This is a part of CTPPS offline software.
+ * Authors:
+ *   Laurent Forthomme (laurent.forthomme@cern.ch)
+ *   Nicola Minafra (nicola.minafra@cern.ch)
+ *
+ ****************************************************************************/
+
+#include "RecoCTPPS/TotemRPLocal/interface/CTPPSDiamondTrackRecognition.h"
+
+#include <cmath>
+#include <cstdio>
+#include <algorithm>
+
+#define PAD_DEFAULT_FUNCTION "(x>[0]-0.5*[1])*(x<[0]+0.5*[1])"
+
+//----------------------------------------------------------------------------------------------------
+
+CTPPSDiamondTrackRecognition::CTPPSDiamondTrackRecognition() :
+  threshold_( 1.5 ), thresholdFromMaximum_( 0.5 ),
+  resolution_( 0.025 ), sigma_( 0.0 ),
+  startFromX_( -2 ), stopAtX_( 38 ),
+  pixelEfficiencyFunction_( PAD_DEFAULT_FUNCTION ),
+  yPosition(0.0), yWidth(0.0), nameCounter(0)
+{}
+
+CTPPSDiamondTrackRecognition::CTPPSDiamondTrackRecognition( const edm::ParameterSet& iConfig ) :
+  threshold_( iConfig.getParameter<double>( "threshold" ) ),
+  thresholdFromMaximum_( iConfig.getParameter<double>( "thresholdFromMaximum" ) ),
+  resolution_( iConfig.getParameter<double>( "resolution" ) ),
+  sigma_( iConfig.getParameter<double>( "sigma" ) ),
+  startFromX_( iConfig.getParameter<double>( "startFromX" ) ),
+  stopAtX_( iConfig.getParameter<double>( "stopAtX" ) ),
+  pixelEfficiencyFunction_( iConfig.getParameter<std::string>( "pixelEfficiencyFunction" ) ),
+  yPosition(0.0), yWidth(0.0), nameCounter(0)
+{
+  if (sigma_==.0) {
+    pixelEfficiencyFunction_ = PAD_DEFAULT_FUNCTION; // simple step function
+  }
+}
+
+//----------------------------------------------------------------------------------------------------
+
+CTPPSDiamondTrackRecognition::~CTPPSDiamondTrackRecognition()
+{}
+
+//----------------------------------------------------------------------------------------------------
+
+void
+CTPPSDiamondTrackRecognition::clear()
+{
+  hitFunctionVectorMap_.clear();
+  mhMap_.clear();
+}
+
+//----------------------------------------------------------------------------------------------------
+
+void
+CTPPSDiamondTrackRecognition::addHit( const CTPPSDiamondRecHit recHit )
+{
+  std::string f_name( "hit_f_" );
+  f_name.append( std::to_string( ++nameCounter ) );
```

Comment:
is ``` nameCounter ``` ever reset? (I'd expect it should be, once per event if not more frequently)